src/transports/auth_negotiate.c

```diff
@@ -14,8 +14,12 @@
 #include "auth.h"
 #include "git2/sys/cred.h"
 
+#ifdef __APPLE__
+#import <GSS/GSS.h>
```

Comment:
So if the user explicitly chose "-DUSE_GSSAPI=gssapi", then we'd still choose the Apple one. I think we need to put backend information into "features.h.in" and use that instead of `__APPLE__`.